nodestream/pipeline/transformers/transformer.py

```diff
@@ -129,3 +131,44 @@ def do_work_on_record(self, record: Any) -> Any:
     @abstractmethod
     async def transform_record(self, record: Any) -> Any:
         raise NotImplementedError
+
+
+class SwitchTransformer(Transformer):
+    __slots__ = (
+        "switch_on",
+        "transformers",
+        "default",
+        "normalization",
+    )
+
+    @staticmethod
+    def guarantee_transformer_from_file_data(file_data):
+        return ClassLoader().load_class(**file_data)
+
+    def __init__(
+        self,
+        switch_on: StaticValueOrValueProvider,
+        cases: Dict[str, dict],
+        default: Dict[str, Any] = None,
+        normalization: Dict[str, Any] = None,
+    ):
+        self.switch_on = ValueProvider.guarantee_value_provider(switch_on)
+        self.transformers = {
+            field_value: self.guarantee_transformer_from_file_data(transformer)
+            for field_value, transformer in cases.items()
+        }
+        self.default = (
+            self.guarantee_transformer_from_file_data(default) if default else None
+        )
+        self.normalization = normalization or {}
```

Comment:
Can you move the initialization logic to `from_file_data`.  The constructor should probably only take initialized class instances. For instance, check out: https://github.com/nodestream-proj/nodestream/blob/main/nodestream/pipeline/transformers/expand_json_field.py#L10